lib/item.py

```diff
@@ -391,17 +405,20 @@ def _castvalue_to_itemtype(self, value, compat):
         """
         # casting of value, if compat = latest
         if compat == ATTRIB_COMPAT_LATEST:
-            mycast = globals()['_cast_' + self._type]
-            try:
-                value = mycast(value)
-            except:
-                logger.warning("Item {}: Unable to cast '{}' to {}".format(self._path, str(value), self._type, str(mycast)))
-                if isinstance(value, list):
-                    value = []
-                elif isinstance(value, dict):
-                    value = {}
-                else:
-                    value = mycast('')
+            if self._type != None:
+                mycast = globals()['_cast_' + self._type]
+                try:
+                    value = mycast(value)
+                except:
+                    logger.warning("Item {}: Unable to cast '{}' to {}".format(self._path, str(value), self._type, str(mycast)))
```

Comment:
The `str(mycast)` is not added to the message since the message contains only 3 tokens to replace.